python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1964 @@
+        table["extra_field_y"].unit = u.deg
+        table["intra_centroid_x"].unit = u.pixel
+        table["intra_centroid_y"].unit = u.pixel
+        table["extra_centroid_x"].unit = u.pixel
+        table["extra_centroid_y"].unit = u.pixel
+        for j in self.config.nollIndices:
+            table[f"Z{j}"].unit = u.um
+
+        self.log.debug("Initialized Zernike table with %d coefficient columns", len(self.config.nollIndices))
+        return table
+
+    def createZkTable(
+        self,
+        extraStamps: DonutStamps,
+        intraStamps: DonutStamps,
+        zkCoeffRaw: np.ndarray,
+        zkCoeffAvg: np.ndarray,
+    ) -> QTable:
+        """Create the Zernike table to store Zernike Coefficients.
+
+        Parameters
+        ----------
+        extraStamps: DonutStamps
+            The extrafocal stamps
+        intraStamps: DonutStamps
+            The intrafocal stamps
+        zkCoeffRaw: np.ndarray
+            Raw zernike coefficients from TARTS
+        zkCoeffAvg: np.ndarray
+            Averaged zernike coefficients
+
+        Returns
+        -------
+        table : `astropy.table.QTable`
+            Table with the Zernike coefficients
+        """
+        zkTable = self.initZkTable()
+        self.log.debug(
+            "Creating Zernike table: intra stamps=%d, extra stamps=%d",
+            len(intraStamps),
+            len(extraStamps),
+        )
+
+        # Add average row
+        row_data = {
+            "label": "average",
+            "used": True,
+            "intra_field_x": np.nan,
+            "intra_field_y": np.nan,
+            "extra_field_x": np.nan,
+            "extra_field_y": np.nan,
+            "intra_centroid_x": np.nan,
+            "intra_centroid_y": np.nan,
+            "extra_centroid_x": np.nan,
+            "extra_centroid_y": np.nan,
+            "intra_mag": np.nan,
+            "extra_mag": np.nan,
+            "intra_sn": np.nan,
+            "extra_sn": np.nan,
+            "intra_entropy": np.nan,
+            "extra_entropy": np.nan,
+            "intra_frac_bad_pix": np.nan,
+            "extra_frac_bad_pix": np.nan,
+            "intra_max_power_grad": np.nan,
+            "extra_max_power_grad": np.nan,
+            "intra_fx": np.nan,
+            "extra_fx": np.nan,
+            "intra_fy": np.nan,
+            "extra_fy": np.nan,
+        }
+
+        # Add Zernike coefficients
+        for i, j in enumerate(self.config.nollIndices):
+            if i < len(zkCoeffAvg):
+                row_data[f"Z{j}"] = zkCoeffAvg[i] * u.um
+            else:
+                row_data[f"Z{j}"] = 0.0 * u.um
+
+        zkTable.add_row(row_data)
+        self.log.debug("Added average row with %d Zernike terms", len(self.config.nollIndices))
+
+        # Add individual donut rows only if we have donut stamps
+        # For TARTS, we typically have one set of coefficients per exposure
+        # We'll create one row per donut if available, otherwise just the
```

Comment:
this is confusing. First because this is for tarts, so the clarification makes it confusing. Aren't you creating this per pair of exposures, even though you then average them later with aggregatorNet?